tools/ncf_doc_rudder.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Usage: ./ncf_doc_rudder.py
+#
+# This is a Python module to generate documentation from generic methods in ncf to be embedded in Rudder manual
+
+import ncf 
+import requests
+import sys
+import re
+from distutils.version import StrictVersion
+from pprint import pprint
+
+# URL of the rudder_info api
+release_info_url = "http://www.rudder-project.org/release-info/rudder/"
+
+def get_min_versions():
+  "Build the dictionnary of minimal Rudder version including compatibility with a given CFEngine version"
+  min_version = {}
+  
+  try:
+    versions = requests.get(release_info_url + "versions").content.decode('ascii').splitlines()
+  except requests.exceptions.RequestException as e:
+    print(e)
+    sys.exit(1)
+
+  for rudder_version in versions:
+    try:
+      cfengine_version = requests.get(release_info_url + "versions/" + rudder_version + "/components/cfengine").content.decode('ascii')
+    except requests.exceptions.RequestException as e:
+      print(e)
+      sys.exit(1)
+    
+    if cfengine_version in min_version:
+      if StrictVersion(rudder_version) >= StrictVersion(min_version[cfengine_version]):
+        break
+    min_version[cfengine_version] = rudder_version
+    
+  return min_version
+
+def rudder_version(min_version, version):
+  """Extract minimal Rudder version"""
+  canonified_version = re.findall(r'>=? ?(\d+\.\d+)', version)[0]
+  
+  cfengine_versions = list(min_version.keys())
+  cfengine_versions.sort(key=StrictVersion)
+  
+  for cfengine_version in cfengine_versions:
+    if StrictVersion(canonified_version) <= StrictVersion(cfengine_version):
+      return min_version[cfengine_version]
+  return min_version[cfengine_versions[-1]]
```

Comment:
I don't get this. It seems to be a default value of "if I don't find any Rudder versions that support CFEngine version N, then return the last Rudder version". But that doesn't make sense - if we have a ncf generic method that only works in CFEngine 3.10, we shouldn't say it works in 3.2 now.